File: game-dart/web/app/util/keyHandler.dart
import 'dart:html';

class KeyHandler {
  Function arrowUpCallback = null;
  Function arrowDownCallback = null;
  Function arrowLeftCallback = null;
  Function arrowRightCallback = null;

  KeyHandler() {
    window.onKeyDown.listen(onKeyDown);
  }

  void onKeyDown(KeyboardEvent event) {
    switch (event.keyCode.toString()) {
      case "38":
        if (arrowUpCallback != null) {
          arrowUpCallback();
        }
        break;
      case "40":
        if (arrowDownCallback != null) {
          arrowDownCallback();
        }
        break;
      case "37":
        if (arrowLeftCallback != null) {
          arrowLeftCallback();
        }
        break;
      case "39":
        if (arrowRightCallback != null) {
          arrowRightCallback();
        }
        break;
      default:
        break;
    }
  }
}

File: game-js/app/util/keyHandler.js
function KeyHandler() {
	this.arrowUpCallback = null;
	this.arrowDownCallback = null;
	this.arrowLeftCallback = null;
	this.arrowRightCallback = null;

	this._start();
}

KeyHandler.prototype._start = function () {
	window.addEventListener('keydown', this._onKeyDown.bind(this));
};

KeyHandler.prototype._onKeyDown = function (event) {
	switch(event.key) {
		case "ArrowUp":
			if(this.arrowUpCallback !== null && this.arrowUpCallback !== undefined) {
				this.arrowUpCallback();
			}
			break;
		case "ArrowDown":
			if(this.arrowDownCallback !== null && this.arrowDownCallback !== undefined) {
				this.arrowDownCallback();
			}
			break;
		case "ArrowLeft":
			if(this.arrowLeftCallback !== null && this.arrowLeftCallback !== undefined) {
				this.arrowLeftCallback();
			}
			break;
		case "ArrowRight":
			if(this.arrowRightCallback !== null && this.arrowRightCallback !== undefined) {
				this.arrowRightCallback();
			}
			break;
		default:
			break;
	}
};